lumicks/pylake/confocal.py

```diff
@@ -136,16 +137,24 @@ def _ordered_axes(self):
     def _to_spatial(self, data):
         raise NotImplementedError
 
-    def _image(self, color):
-        if color not in self._cache:
-            photon_counts = getattr(self, f"{color}_photon_count").data
-            self._cache[color] = reconstruct_image_sum(photon_counts, self.infowave.data, self._shape)
-            self._cache[color] = self._to_spatial(self._cache[color])
-        return self._cache[color]
+    @cachetools.cachedmethod(lambda self: self._cache)
+    def _image(self, channel):
+        channel_data = getattr(self, f"{channel}_photon_count").data
+        raw_image = reconstruct_image_sum(channel_data, self.infowave.data, self._shape)
+        return self._to_spatial(raw_image)
 
-    def _timestamps(self, sample_timestamps):
-        timestamps = reconstruct_image(sample_timestamps, self.infowave.data, self._shape, reduce=np.mean)
-        return self._to_spatial(timestamps)
+    @cachetools.cachedmethod(lambda self: self._cache)
+    def _timestamps(self, channel):
+        assert channel == "timestamps"
+        # Uses the timestamps from the first non-zero-sized photon channel
+        for channel_data in (getattr(self, f"{color}_photon_count").timestamps 
+                                for color in ("red", "green", "blue")):
```

Comment:
This looks pretty messy to me. Consider one of these two alternatives:
```suggestion
        for color in "red", "green", "blue":
            channel_data = getattr(self, f"{color}_photon_count").timestamps
```
```suggestion
        channels = (getattr(self, f"{color}_photon_count").timestamps 
                    for color in "red", "green", "blue")
        for channel_data in channels:
```